buildSrc/build.gradle.kts

```diff
@@ -0,0 +1,17 @@
+import java.util.Properties
+
+plugins {
+    `kotlin-dsl`
+}
+
+repositories {
+    mavenCentral()
+}
+
+val kotlinVersion = file("../gradle.properties").inputStream().use {
+    Properties().apply { load(it) }
+}.getProperty("kotlinVersion") ?: throw IllegalStateException("Property 'kotlinVersion' must be defined in ../gradle.properties")
```

Comment:
This approach significantly limits the ways how `kotlinVersion` can be defined. Due to that we'd like to avoid having the code that depends on kotlin plugin in buildSrc, especially considering that it is only needed in one subproject so far.

Reply:
I totally agree, this was actually the most annoying part. I just wished that the `buildSrc` project had access to the properties of the root project, but that is unfortunately not the case. 

Anyway, I'll move the relevant code to an (simplified) inline block inside the `core` subproject, since that is the only place that uses this.